budget/app.py

```diff
@@ -1,14 +1,176 @@
 import json
 import logging
 
+import boto3
+from botocore.exceptions import ClientError
+import requests
+
 log = logging.getLogger(__name__)
 log.setLevel(logging.DEBUG)
 
+# TODO configure this URL?
+SYNAPSE_TEAM_MEMBER_URL = 'https://repo-prod.prod.sagebase.org/repo/v1/teamMembers'
+
+# TODO pull this config from S3?
+USER_BUDGET_BY_TEAM = {
+  '12345': {
+    'amount': 2000,
+    'period': 'annual'
+  },
+  '67890': {
+    'amount': 100,
+    'period': 'annual'
+  }
+}
+
+
+def get_client(service):
+  return boto3.client(service)
+
+
+def get_users_by_team(teams):
+  '''Get users from synapse teams
+
+  Returns a dictionary of users with a list of their team memberships
+  '''
+  teams_by_user_id = {}
+  for team_id in teams:
+    synapse_url = f'{SYNAPSE_TEAM_MEMBER_URL}/{team_id}'
+    response = requests.get(synapse_url)
```

Comment:
This will return a *page* of team members, not *all* team members.   (See https://rest-docs.synapse.org/rest/GET/teamMembers/id.html)  SynapseClient.getTeamMembers() returns a *generator* that will iterate through all team members.

Reply:
Good point. I am making a note in my JIRA issue to deal with the pagination in a future PR.